src/core/consumable.cpp

```diff
@@ -0,0 +1,36 @@
+#include "core/consumable.h"
+
+Consumable::Consumable(bool active/*=false*/)
```

Comment:
that's a terrible name (it's usually used for physical goods, food products, fuel etc.)
trigger? one-shot action?
SetConsumed: likewise.